packages/cadl-ranch-specs/http/azure/core/lro/rpc/main.tsp

```diff
@@ -0,0 +1,98 @@
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/cadl-ranch-expect";
+import "@typespec/rest";
+import "@typespec/versioning";
+
+using Azure.Core;
+using global.Azure.Core.Traits;
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+
+#suppress "@azure-tools/typespec-azure-core/casing-style" "For spec"
+@doc("Illustrates bodies templated with Azure Core with long-running RPC operation")
+@scenarioService(
+  "/azure/core/lro/rpc",
+  {
+    versioned: Versions,
+  }
+)
+namespace _Specs_.Azure.Core.Lro.Rpc;
+
+@doc("The API version.")
+enum Versions {
+  @doc("The 2022-12-01-preview version.")
+  @useDependency(global.Azure.Core.Versions.v1_0_Preview_2)
+  v2022_12_01_preview: "2022-12-01-preview",
+}
+
+@doc("Options for the generation.")
+model GenerationOptions {
+  @doc("Prompt.")
+  prompt: string;
+}
+
+model GenerationResponse is global.Azure.Core.Foundations.OperationStatus<GeneratonResult>;
+// fix warning in Azure.Core.Foundations.OperationStatus
+@@visibility(global.Azure.Core.Foundations.OperationStatus.id, "read");
+
```

Comment:
I am been lazy here, re-using the Foundations model.

If anyone still prefer to have an explicit response model, I can write it explicitly (it is still just `id+status+result+error`).